.github/workflows/cherrypick-to-beta.yaml

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+name: Cherry-pick Labeled PR to Beta Branch
+
+on:
+  pull_request:
+    branches: master
+    types: [labeled]
+
+permissions: write-all
+
+jobs:
+  cherrypick_to_beta:
```

Comment:
Instead of duplicating this in the framework and engine, we should create the action in Cocoon. We can then publish it, and install on the framework and engine repos. That will reduce the maintenance cost as we'll only need to change one place.

Reply:
right, umm but

1. the trigger of action (e.g. `on: pullrequest`) comes from the repository where the github action lives.  if we put it on the cocoon repository, it will trigger from cocoon prs instead of framework/engine prs.<br >
to trigger an action from another repository different from where the action lives, we could theoretically use the [custom dispatch event](https://github.com/orgs/community/discussions/26323), but it requires us to hand craft the event payload ourselves (which would not be automated unless we create another action on the triggering repo itself), and also needs embedding of extra tokens.

3. To host both framework and engine in one script, we need to support the selection of beta / stable branch, engine / flutter repository, master / main branch etc., which requires a number of parallel `if` steps in one workflow. ( e.g. two neighboring steps A and B where if framework, do step A. if not, do step B). My understanding is that separating them out would maybe be clearer?

Reply:
1. See https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace. We have existing Github actions that are triggered based on actions that are installed from the marketplace
2. We can reduce the nesting here to a single if statement, which shouldn't be an issue

Reply:
1. umm true, if we plan to go that way I think it would require some bots/author in cocoon to authroize itself to open pull request in framework / engine repo (if we open pr from within framework itself the authorization comes for free). maybe reusing auto submit bot?
<br > However I do prefer per repository action over cross repo action, cause I think the extra cross-repo logic could introduce overhead and be harder to debug the components of marketplace action when it hits an error. (e.g. a readfile marketplace with 100 stars uses more space than a curl command that fetches the file content). 

3. umm to use a single if statement, does it mean having the entire framework job and the entire engine job in one file (e.g. a 200 line file, 100 line under if statement for framework workflow, and 100 line for engine workflow) My understanding is that would be similar to having 2 separate files?

3. right I read the link Casey shared of https://docs.google.com/presentation/d/1C_uhRjwWnjBx82JQmTYghbBbrjFGtupRCETTOEw3rzs/edit#slide=id.p, I will try to finish the design doc first.